dds/DCPS/transport/framework/TransportReassembly.cpp

```diff
@@ -403,10 +402,10 @@ void TransportReassembly::check_expirations(const MonotonicTimePoint& now)
                      "purge expired leaving %B fragments\n", fragments_.size()));
         }
       } else {
-        expiration_queue_.insert(std::make_pair(iter->second.expiration_, pos->second));
+        expiration_queue_.push_back(std::make_pair(iter->second.expiration_, iter->first));
```

Comment:
Since push_back doesn't sort like insert, I think we need to be be careful with our while loop's check for `expiration_queue_.front().first <= now`. If the expiration is only ever set to "now + configured_expiration" then I think it's safe, but if we did anything fancier with the expirations (e.g. if "initial" expiration > "update" expiration) they could potentially get inserted out of order and the loop's end condition would be incorrect.

Reply:
Viewed another way, we would just be holding on to fragments longer than anticipated.